monkey/monkey_island/cc/ui/src/components/pages/MarketplacePage.js

```diff
@@ -32,41 +31,15 @@ const a11yProps = (index) => {
 }
 
 const MarketplacePage = () => {
-  const [availablePlugins, setAvailablePlugins] = useState([]);
-  const [installedPlugins, setInstalledPlugins] = useState([]);
-  const [tabValue, setTabValue] = useState(0);
-  const authComponent = new AuthComponent({});
-
-  useEffect(() => {
-    refreshAvailablePlugins();
-  }, []);
+  const {numberOfPluginsThatRequiresUpdate} = useContext(PluginsContext);
 
-  useEffect(() => {
-    refreshInstalledPlugins();
-  }, []);
-
-  const refreshAvailablePlugins = (forceRefresh: boolean = false) => {
-    let url = '/api/agent-plugins/available/index';
-    if (forceRefresh) {
-      url += '?force_refresh=true';
-    }
-    authComponent.authFetch(url, {}, true).then(res => res.json()).then(plugins => {
-      setAvailablePlugins(plugins.plugins);
-    });
-  };
-
-  const refreshInstalledPlugins = () => {
-    authComponent.authFetch('/api/agent-plugins/installed/manifests', {}, true).then(res => res.json()).then(plugins => {
-      setInstalledPlugins(plugins);
-    });
-  };
```

Comment:
Extract these methods and use them in main + pass it down through context so that children can use it